src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SkeletonReferenceSet.cs

```diff
@@ -60,9 +63,7 @@ private static (AssemblyMetadata, ISupportDirectMemoryAccess?) ComputeMetadata(I
                 // this is unfortunate that if we give stream, compiler will just re-copy whole content to 
                 // native memory again. this is a way to get around the issue by we getting native memory ourselves and then
                 // give them pointer to the native memory. also we need to handle lifetime ourselves.
-                var metadata = AssemblyMetadata.Create(ModuleMetadata.CreateFromImage(supportNativeMemory.GetPointer(), (int)stream.Length));
-
-                return (metadata, supportNativeMemory);
+                return AssemblyMetadata.Create(ModuleMetadata.CreateFromImage(supportNativeMemory.GetPointer(), (int)stream.Length, owner: supportNativeMemory));
```

Comment:
Another approach would be to use UnmanagedMemoryStream instead of ISupportDirectMemoryAccess. The compiler can check if given Stream is UnmanagedMemoryStream and implement this logic in the compiler.

Reply:
Let me try taht.  No promises that i'll succeed :D

Reply:
@tmat i think i understood what you meant.  can you look and see if hte PR now fits what you were thinking vis-a-vis UnmanagedMemoryStream?